core/src/test/java/org/elasticsearch/index/query/GeoShapeQueryBuilderTests.java

```diff
@@ -265,7 +265,11 @@ public void testWrongFieldType() throws IOException {
         ShapeBuilder shape = RandomShapeGenerator.createShapeWithin(random(), null, shapeType);
         final GeoShapeQueryBuilder queryBuilder = new GeoShapeQueryBuilder(STRING_FIELD_NAME, shape);
         QueryShardException e = expectThrows(QueryShardException.class, () -> queryBuilder.toQuery(createShardContext()));
-        assertThat(e.getMessage(), containsString("Field [mapped_string] is not of type [geo_shape] but of type [text]"));
+        if (getCurrentTypes().length == 0) {
```

Comment:
I don't understand why this can be true when in L263 we use `assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0)`
I investigated this error yesterday with Jim and he pushed the "assumeTrue" part, so this test shouldn't run in those cases. Did you run the mentioned reproduction line on current master, because I could get it to fail using it on 77617c8e62f7fa1407c62bdf33e4892206d62952